### PR TITLE
chore(ci): Fix application of `rustflags`

### PR DESCRIPTION
**Pull request by bruceg**

### CODE REVIEW

File: scripts/environment/bootstrap-ubuntu-20.04.sh [hunk 1]
```diff
@@ -130,13 +136,13 @@ tar \
...
-CARGO_OVERRIDE_DIR="${HOME}/.csh check for exisargo"
+CARGO_OVERRIDE_DIR="${HOME}/.cargo"
 mkdir -p "${CARGO_OVERRIDE_DIR}"
 
 CARGO_OVERRIDE_CONF="${CARGO_OVERRIDE_DIR}/config.toml"
-cat <<EOF >"${CARGO_OVERRIDE_CONF}"
+cat <<EOF >>"${CARGO_OVERRIDE_CONF}"
 [target.x86_64-unknown-linux-gnu]
-rustflags = ["-C", "linker=clang", "-C", "link-arg=-fuse-ld=/usr/bin/mold"]
+rustflags = ["-D", "warnings", "-C", "linker=clang", "-C", "link-arg=-fuse-ld=/usr/bin/mold"]
```
Comment (spencergilbert) [inline scripts/environment/bootstrap-ubuntu-20.04.sh:145]: Do we need to update the rustflags for other targets here as well?

Comment (tobz) [reply scripts/environment/bootstrap-ubuntu-20.04.sh:145]: We only run our primary CI suite against x86-64 Linux, so Bruce is just mirroring his removal of `-D warnings` from the `RUSTFLAGS` environment variable we were setting directly in `.github/workflows/test.yaml`, etc.

**Review by spencergilbert — APPROVED** [review]

**Review by tobz — APPROVED** [review]
Thank you for noticing, and fixing, my blunder with setting the linker in CI. 😓

File: scripts/environment/bootstrap-ubuntu-20.04.sh [hunk 2]
```diff
@@ -1,6 +1,12 @@
 #! /usr/bin/env bash
 set -e -o verbose
 
+if [ -n "$RUSTFLAGS" ]
```
Comment (jszwedko) [inline scripts/environment/bootstrap-ubuntu-20.04.sh:4]: 👍

File: scripts/environment/bootstrap-ubuntu-20.04.sh [hunk 3]
```diff
@@ -130,13 +136,13 @@ tar \
 cp "${TEMP}/${MOLD_TARGET}/bin/mold" /usr/bin/mold
 
 # Set Cargo to use mold as its linker.
-CARGO_OVERRIDE_DIR="${HOME}/.csh check for exisargo"
```
Comment (jszwedko) [inline scripts/environment/bootstrap-ubuntu-20.04.sh:133]: 😂

**Review by jszwedko — APPROVED** [review]
Sweet! I see it flagged a `println!` so seems to be working.

**Comment by bruceg** [comment]
This appears to be allowing warnings to pass CI, so something else is needed here.

**Review by jszwedko — COMMENTED** [review]
Looks like Rust docs use a different `rustflags` configuration: `rustdocflags`.

I'd still be ok with having this go in as an incremental improvement though and follow up on the rust docs tests flags.

**Pull request merged**